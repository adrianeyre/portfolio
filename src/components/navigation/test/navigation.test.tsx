/**
 * @jest-environment jsdom
 */
import { render } from '@testing-library/react';


import INavigationProps from '../interface/navigation-props';
import Navigation from '../navigation';

describe('Navigation', () => {
	const mockRouteProp: any = jest.fn();

	const defaultProps: INavigationProps = {
		data: undefined,
		linksData:  undefined,
		scrollToAnchor: mockRouteProp,
		showModal: mockRouteProp,
	};

	it('Should render correctly', () => {
		const navigationComponent = render(<Navigation { ...defaultProps }/>);

		expect(navigationComponent).toMatchSnapshot();
	});
});
