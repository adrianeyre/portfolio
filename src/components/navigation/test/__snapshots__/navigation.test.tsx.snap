// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navigationContainer"
      >
        <nav
          class="navbar navbar-expand-lg navbar-light bg-light fixed-top"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            ADRIAN EYRE
          </a>
          <span
            class="topBarLinks"
          >
            <div
              class="linksContainer"
            />
          </span>
          <button
            aria-controls="basic-navbar-nav"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="basic-navbar-nav"
          >
            <div
              class="mr-auto navbar-nav"
            >
              <a
                class="nav-link"
                href="#"
                role="button"
                tabindex="0"
              >
                <div
                  class="linksContainer"
                />
              </a>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="navigationContainer"
    >
      <nav
        class="navbar navbar-expand-lg navbar-light bg-light fixed-top"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          ADRIAN EYRE
        </a>
        <span
          class="topBarLinks"
        >
          <div
            class="linksContainer"
          />
        </span>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            class="mr-auto navbar-nav"
          >
            <a
              class="nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              <div
                class="linksContainer"
              />
            </a>
          </div>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
